### PR TITLE
Add np12

### PR DESCRIPTION
**Pull request by nut1414**
## Description

Adding np12 keypad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by ridingqwerty — APPROVED** [review]

File: keyboards/np12/readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# np12
+
+![np12](https://i.imgur.com/gci7qqZ.jpg)
+
+![np12 pcb](https://i.imgur.com/DIIvW95.jpg)
```
Comment (fauxpark) [inline keyboards/np12/readme.md:5]:
```suggestion
![np12](https://i.imgur.com/gci7qqZh.jpg)

![np12 pcb](https://i.imgur.com/DIIvW95h.jpg)
```

File: keyboards/np12/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+* Keyboard Maintainer: [nut1414](https://github.com/nut1414)
+* Hardware Supported: np12 keypad
+* Hardware Availability: Open Source https://github.com/nut1414/np12
+
+
+Make example for this keypad (after setting up your build environment):
+
+    make np12:default
+	
+
```
Comment (fauxpark) [inline keyboards/np12/readme.md:18]:
```suggestion
```

File: keyboards/np12/rules.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no		# Console for debug
+COMMAND_ENABLE = no		# Commands for debug and configuration
+SLEEP_LED_ENABLE = no	# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no		# Audio output
+RGBLIGHT_ENABLE = no	# Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/np12/rules.mk:19]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by nut1414** [comment]
Thanks for reviewing this.

**Review by drashna — APPROVED** [review]

**Pull request merged**